crates/storage/provider/src/test_utils/mock.rs

```diff
@@ -164,6 +164,26 @@ impl StateCommitmentProvider for MockEthProvider {
     type StateCommitment = <MockNode as NodeTypes>::StateCommitment;
```

Comment:
```suggestion
impl<T> StateCommitmentProvider for MockEthProvider<T> {
    type StateCommitment = <MockNode as NodeTypes>::StateCommitment;
```
this is instead fixed by generalising the impls. in some cases you may need trait bound `T: reth_primitive_traits::SignedTransaction`

Reply:
this change is what we want to upstream, but I'm merging this pr into the base branch anyway to unblock

Reply:
opened an issue https://github.com/paradigmxyz/reth/issues/14038